test/common/access_log/access_log_manager_impl_test.cc

```diff
@@ -12,34 +12,326 @@
 #include "gmock/gmock.h"
 
 using testing::_;
+using testing::NiceMock;
 using testing::Return;
+using testing::ReturnNew;
 using testing::ReturnRef;
+using testing::Sequence;
 
 namespace Envoy {
 namespace AccessLog {
 
-TEST(AccessLogManagerImpl, reopenAllFiles) {
-  Api::MockApi api;
-  Filesystem::MockInstance file_system;
-  EXPECT_CALL(api, fileSystem()).WillRepeatedly(ReturnRef(file_system));
-  Event::MockDispatcher dispatcher;
-  Thread::MutexBasicLockable lock;
-
-  std::shared_ptr<Filesystem::MockFile> log1(new Filesystem::MockFile());
-  std::shared_ptr<Filesystem::MockFile> log2(new Filesystem::MockFile());
-  AccessLogManagerImpl access_log_manager(api, dispatcher, lock);
-  EXPECT_CALL(file_system, createFile("foo", _, _)).WillOnce(Return(log1));
-  access_log_manager.createAccessLog("foo");
-  EXPECT_CALL(file_system, createFile("bar", _, _)).WillOnce(Return(log2));
-  access_log_manager.createAccessLog("bar");
+class AccessLogManagerImplTest : public TestBase {
+protected:
+  AccessLogManagerImplTest()
+      : file_(new NiceMock<Filesystem::MockFile>), thread_factory_(Thread::threadFactoryForTest()),
+        access_log_manager_(timeout_40ms_, api_, dispatcher_, lock_, store_) {
+    EXPECT_CALL(file_system_, createFile("foo"))
+        .WillOnce(Return(ByMove(std::unique_ptr<NiceMock<Filesystem::MockFile>>(file_))));
+
+    EXPECT_CALL(api_, fileSystem()).WillRepeatedly(ReturnRef(file_system_));
+    EXPECT_CALL(api_, threadFactory()).WillRepeatedly(ReturnRef(thread_factory_));
+  }
+
+  NiceMock<Api::MockApi> api_;
+  NiceMock<Filesystem::MockInstance> file_system_;
+  NiceMock<Filesystem::MockFile>* file_;
+  const std::chrono::milliseconds timeout_40ms_{40};
+  Stats::IsolatedStoreImpl store_;
+  Thread::ThreadFactory& thread_factory_;
+  NiceMock<Event::MockDispatcher> dispatcher_;
+  Thread::MutexBasicLockable lock_;
+  AccessLogManagerImpl access_log_manager_;
+};
+
+TEST_F(AccessLogManagerImplTest, BadFile) {
+  EXPECT_CALL(dispatcher_, createTimer_(_));
+  EXPECT_CALL(*file_, open_()).WillOnce(Return(Api::SysCallBoolResult{false, 0}));
+  EXPECT_THROW(access_log_manager_.createAccessLog("foo"), EnvoyException);
+}
+
+TEST_F(AccessLogManagerImplTest, flushToLogFilePeriodically) {
+  NiceMock<Event::MockTimer>* timer = new NiceMock<Event::MockTimer>(&dispatcher_);
```

Comment:
consider using SimulatedTimeSystem rather than mocking timers. It might be a cleaner test. It didn't *look* like they were carried over from master.

Reply:
These tests were carried over from master, e.g.: https://github.com/envoyproxy/envoy/blob/6c95ed9cbfa5e104208c951271817e487639574c/test/common/filesystem/filesystem_impl_test.cc#L127